compyle/cython_generator.py

```diff
@@ -90,6 +90,127 @@ def __init__(self, **kwargs):
                         methods=self.methods)
 
 
+def condense_multiline_calls(lines:list):
+    """
+    This function takes a list of lines and condenses multiline calls into
+    single lines, while preserving indentation, and moving any inline comments
+    to the end of the line.
+
+    Parameters
+    ----------
+    lines : list of str
+        The lines of code to condense.
+    
+    Returns
+    -------
+    condensed_lines : list of str
+        The condensed lines of code.
+    """
+    def _line_is_complete(line:str):
+        """
+        This function checks if a line is complete, i.e. if it has the same
+        number of opening and closing parentheses, and if it does not end with
+        a backslash.
+
+        Parameters
+        ----------
+        line : str
+            The line to check.
+        
+        Returns
+        -------
+        is_complete : bool
+            True if the line is complete, False otherwise.
+        """
+        tmp_line = line.strip()
+        cond1 = tmp_line.count('(') == tmp_line.count(')')
+        cond2 = not tmp_line.endswith('\\')
+        return (cond1 and cond2)
+    
+    lines = [line.rstrip() for line in lines]
+    condensed_lines = []
+    line_count = 0
+    for idx, line in enumerate(lines):
+        if line.endswith('\n'):
+            # Remove the newline character
+            line = line[:-1]
+
+        if line_count > idx:
+            # This line has already been condensed
+            continue
+        
+        cond = (line.count('(') == 0) and (line.count(')') >= 1)
+        if _line_is_complete(line) or cond:
+            # This line is complete, or it only has closing parentheses
+            condensed_lines.append(line)
+            line_count += 1
+            continue
+        else:
+            # This line is incomplete, so it needs to be condensed
+            current_line = line.rstrip()
+            if current_line.endswith('\\'):
+                # Remove the backslash
+                current_line = current_line[:-1]
+            
+            if idx + 1 >= len(lines):
+                # This line is incomplete, but there are no more lines
+                raise SyntaxError(f"{idx}: {line}\n '(' was never closed")
+            
+            for next_line in lines[idx+1:]:
+                # Iterate over the next lines, adding them to the current line
+                # until the line is complete
+                tmp = current_line.split('#')
```

Comment:
One issue here with more generic Python is if you have a "#" inside a string. However, that seems unlikely in compyle. In any case the current Cython code is quite fragile and needs an overhaul to use the ast instead of crude string manipulations so perhaps this is OK for now.